### PR TITLE
[4.0] HTML on Locked-Article

### PR DESCRIPTION
**Pull request by brianteeman**
Still a work in progress

Currently working for checked out
but as a side effect it is also working elsewhere - that is to be resolved hence its a wip

### CODE REVIEW

File: libraries/cms/html/jgrid.php [hunk 1]
```diff
@@ -78,11 +78,12 @@ public static function action($i, $task, $prefix = '', $active_title = '', $inac
 			{
 				$html[] = ' href="javascript:void(0);" onclick="return Joomla.listItemTask(\'' . $checkbox . $i . '\',\'' . $prefix . $task . '\')"';
 			}
-
-			$html[] = $tip ? ' title="' . $title . '"' : '';
+
```
Comment (Quy) [inline libraries/cms/html/jgrid.php]: Remove tabs.

File: libraries/cms/html/jgrid.php [hunk 2]
```diff
@@ -62,12 +62,12 @@ public static function action($i, $task, $prefix = '', $active_title = '', $inac
 		{
 			$title = $enabled ? $active_title : $inactive_title;
 			$title = $translate ? Text::_($title) : $title;
-			$title = HTMLHelper::_('tooltipText', $title, '', 0);
+			$ariaid  = $checkbox . $task . $i . '-desc';
```
Comment (Quy) [inline libraries/cms/html/jgrid.php]:
```suggestion
			$ariaid = $checkbox . $task . $i . '-desc';
```

**Comment by Quy** [comment]
The tooltip displays at the top and not next to the lock icon.

Windows 10, Firefox 66.0.5

locked

**Comment by richard67** [comment]
Can confirm Quy's result.

**Comment by brianteeman** [comment]
Can you retest please - I believe I have fixed the styling issue

**Comment by Quy** [comment]
Now there is an issue with the search box.
search

**Comment by brianteeman** [comment]
Will have to leave it to someone who understands css to fix ;)

**Comment by Quy** [comment]
Add after 
```
[id="filter[search]-desc"] {
  top: 100%;
}
```

File: administrator/templates/atum/scss/blocks/_form.scss [hunk 3]
```diff
@@ -124,8 +124,8 @@ legend {
 }
 
  /* reveal associated tooltip on focus */
-:focus + [role="tooltip"] {
+:focus + [role="tooltip"],
+:hover + [role="tooltip"] {
   display: block;
   position: absolute;
-  bottom: 100%;
 }
```
Comment (Quy) [inline administrator/templates/atum/scss/blocks/_form.scss]:
```suggestion
}

[id="filter[search]-desc"] {
  top: 100%;
}
```

Comment (brianteeman) [reply administrator/templates/atum/scss/blocks/_form.scss]: I had hope to avoid field specific css - but I guess that's not going to be possible

**Comment by richard67** [comment]
Tool tips for checked out icon and search box are placed well now.

But there is an issue with the tool tip for the published icon:

Unbenannt

Placement is good, but colours are strange.

**Comment by brianteeman** [comment]
@richard67 
> Currently working for checked out
but as a side effect it is also working elsewhere - that is to be resolved hence its a wip

**Comment by richard67** [comment]
@brianteeman I see. So shall we wait with testing as long as a wip? Or test this PR and the rest will be done with another, new one?

**Comment by brianteeman** [comment]
@richard67 I cant replicate that tooltip on the publish icon - I dont get anything at all on hover

**Comment by richard67** [comment]
@brianteeman Without this PR, I also get nothing on hover at the publish icon, but with this PR it get as shown in the screenshot above. With PR means: Clean install on current 4.0-dev with blog sampledata, then applied PR on current 4.0-dev using patchtester, then `npm install`, then forced page reload (shoft + reload in Firefox) in order not to use files from browser cache. Just tried it again with new clean install + blog sample data.

**Comment by richard67** [comment]
@Quy Can you check if you have strange toop tip on published icon with this PR (see also my comments above)?

**Comment by Quy** [comment]
Same issue. It needs to be updated from popover to tooltip.

tooltip

**Comment by Quy** [comment]
PR #25161 fixes the popover display issue or popover not displaying.

**Comment by richard67** [comment]
@Quy So this PR here is ok for the checked out icon and the search box, and PR #25161 fixes what we saw here with the publish icon?

**Comment by Quy** [comment]
Yes. Eventually, Brian wants to convert the popover to make it a11y.

**Comment by brianteeman** [comment]
small steps are easier to test than one massive pull request

**Comment by richard67** [comment]
Not sure if I understand yet. So this PR here is ok?

**Comment by Quy** [comment]
Both PRs are required.

**Comment by richard67** [comment]
Hmm, I see. Have to merge that manually then, patch tester detects conflict.

**Comment by Quy** [comment]
Yes. I manually patched the other PR to this PR.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 8498942a43e7d3e0d50efbd2a5c2cd832d2341e1<br /><br />Tooltip on the publish icon was not subject of this PR, it should be done with another one.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24899">issues.joomla.org/tracker/joomla-cms/24899</a>.</sub>

**Comment by richard67** [comment]
@Quy It seems Brian wants to do the other tool tip (publish icon) with another PR, so I tested this one here ok.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8498942a43e7d3e0d50efbd2a5c2cd832d2341e1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24899">issues.joomla.org/tracker/joomla-cms/24899</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24899">issues.joomla.org/tracker/joomla-cms/24899</a>.</sub>

**Comment by roland-d** [comment]
@brianteeman Could you please fix the conflict so I can merge it? Due to PR #25161 being merged this file is in conflict.

**Comment by brianteeman** [comment]
resolved conflicts

**Comment by roland-d** [comment]
Thanks although I am not a fan of the black with strong white text :) That is not part of this PR anyway.

**Comment by brianteeman** [comment]
@roland-d neither am I ;)

**Pull request merged**